components/CheckBox/CheckBox.tsx

```diff
@@ -54,62 +59,169 @@ function CheckBox({ filters, handleFilters }) {
   };
 
   return (
-    <Collapse
-      className={styles.checkbox}
-      bordered={false}
-      expandIcon={({ isActive }) => (
-        <CaretRightOutlined rotate={isActive ? 90 : 0} />
-      )}
-    >
-      <Panel header="Filters" key={0}>
-        {semesters.map((b, index1) => (
-          <Collapse
-            style={{ background: "white" }}
-            bordered={false}
-            expandIcon={({ isActive }) => (
-              <CaretRightOutlined rotate={isActive ? 90 : 0} />
+    <div className={styles.layer}>
+      {/* LEI */}
+
+      <Collapse
+        className={styles.checkbox}
+        bordered={false}
+        expandIcon={({ isActive }) => (
+          <CaretRightOutlined rotate={isActive ? 90 : 0} />
+        )}
+      >
+        <Panel header="LEI" key={0}>
+          {lei.map((b, index1) => (
+            <Collapse
+              style={{ background: "white" }}
+              bordered={false}
+              expandIcon={({ isActive }) => (
+                <CaretRightOutlined rotate={isActive ? 90 : 0} />
+              )}
+              key={index1}
+            >
+              <Panel header={b} key={index1 + 1}>
+                <div style={{ fontWeight: 400 }}>
+                  {/* <div>
+                    <Checkbox type="checkbox" onChange={}>
+                      {" "}
+                      Check all
+                    </Checkbox>
+                  </div> */}
+                  {event[index1]?.map(
+                    (
+                      value: {
+                        id: number;
+                        name: string;
+                        groupId: number;
+                        semester: number;
+                      },
+                      index: number
+                    ) => (
+                      <React.Fragment key={index}>
+                        <div>
+                          <Checkbox
+                            onChange={() => handleToggle(value.id)}
+                            type="checkbox"
+                            checked={
+                              Checked.indexOf(value.id) === -1 ? false : true
+                            }
+                          >
+                            {value.name}
+                          </Checkbox>
+                        </div>
+                      </React.Fragment>
+                    )
+                  )}
+                </div>
+              </Panel>
+            </Collapse>
+          ))}
+        </Panel>
+      </Collapse>
```

Comment:
You are doing some repeated code regarding this `Collapse`. This should be refactored to a component that receives some props (the things that differ from lei, mei and others) 🙏

Reply:
That would be difficult since they don't all follow the same format. Others, for example, doesn't have any sub-categories which means it only has one `Collapse` element and not two nested ones like LEI and MEI. Also, on the schedules page, the MEI filter only has the `1st semester`and `2nd semester` sub-sub-categories for the first year, so the second year has one less `Collapse` component.

Either way, I made some improvements on the mapping on `CheckBox.tsx` so I don't repeat the same code for LEI and MEI. Check it out :pray: